### ISSUE TITLE
How to approach training fairseq RoBERTa for STS-B 

### ISSUE THREAD

**Issue by 123epsilon** [pytorch/fairseq#3139]
#### What is your question?
I am trying to train the fairseq RoBERTa model on the STS-B dataset. What is the best way to go about this? I am having some confusion using fairseq-train as it seems to be geared towards classification tasks and the STS data is of the form 
(sentence, sentence, rating) where rating, the label, is a float from [0,5]. This lends itself to a regression approach but I am unfamiliar with the fairseq library.

I thought about normalizing the data by dividing by 5.0, placing it on a scale from [0,1] then train a new head using only one class. However, I am quite sure that fairseq expects the label to be a class index and not a float.

Other options could be rounding the values to their closest integer neighbor and create categories that way.

Essentially: How should I format the labels for STS-B for training with the RoBERTa model? 

#### What's your environment?

 - fairseq Version (e.g., 1.0 or master): installed from git (git+https://github.com/pytorch/fairseq.git@4a6f89d373dafc50b416092b41c070c304b31698)
 - PyTorch Version (e.g., 1.0): 1.5.0
 - OS (e.g., Linux): Windows 10
 - How you installed fairseq (`pip`, source): Git
 - Build command you used (if compiling from source): n/a
 - Python version: 3.7.4
 - Any other relevant information:

**Comment by chinmaydd** [comment]
Hey everyone, thanks for the awesome project ! I am running into this same issue as well. Would you be able to help us with a workaround or a potential solution?

Thanks!

**Comment by chinmaydd** [comment]
Hey @123epsilon, not sure if you have solved this issue already but just adding a potential resolution in case anyone else faces this in the future. This is probably caused because target labels for samples could not be loaded from the data folder.

The regression task for `sentence_prediction` requires data to be loaded in a different way, specifically label files should be present as : `{split.label}` (for instance `train.label` specifying labels for training data) in the `label` folder. This file should have target labels, each of them on a new line, in a raw format and not in the binarized format as is required by other tasks. 

Please refer [Line 222](https://github.com/pytorch/fairseq/blob/master/fairseq/tasks/sentence_prediction.py#L222) of `sentence_prediction.py`. I hope this helps!
